src/orca/services/sevenbridges/hook.py

```diff
@@ -0,0 +1,48 @@
+from functools import cached_property
+
+from airflow.hooks.base import BaseHook
+from sevenbridges import Api
+
+from orca.services.sevenbridges.client_factory import SevenBridgesClientFactory
+from orca.services.sevenbridges.tasks import SevenBridgesTasks
+
+
+class SevenBridgesHook(BaseHook):
+    """Wrapper around SevenBridges client and tasks classes.
+
+    This hook was inspired by the Asana Airflow provider package:
+    https://github.com/apache/airflow/blob/main/airflow/providers/asana/hooks/asana.py
+    """
+
+    conn_name_attr = "sbg_conn_id"
+    default_conn_name = "sbg_default"
+    conn_type = "sbg"
+    hook_name = "SevenBridges"
+
+    def __init__(self, conn_id: str = default_conn_name, *args, **kwargs):
+        """Construct SevenBridgesHook using an Airflow connection.
+
+        Args:
+            conn_id: An Airflow connection ID.
+                Defaults to ``default_conn_name``.
+        """
+        super().__init__(*args, **kwargs)
+        self.connection = self.get_connection(conn_id)
+        extras = self.connection.extra_dejson
+        self.project = extras.get("project")
+
+    def get_conn(self) -> Api:
+        """Retrieve authenticated SevenBridges client."""
+        return self.client
+
+    @cached_property
+    def client(self) -> Api:
+        """Retrieve authenticated SevenBridges client."""
+        client_args = SevenBridgesClientFactory.map_connection(self.connection)
+        factory = SevenBridgesClientFactory(**client_args)
+        return factory.get_client()
+
+    @cached_property
+    def tasks(self) -> SevenBridgesTasks:
```

Comment:
I think this makes sense.  Just to make sure I'm on the same page, the hook uses the `SevenBridgesClientFactory` to create the client, then the `SevenBridgesTasks` itself takes in the client and has all the commands that the sbg client can do on a specific project.

Is there a way to create projects?

Reply:
Looks like there is a way to create projects in the `sevenbridges` [python package](https://sevenbridges-python.readthedocs.io/en/latest/sevenbridges.models/?highlight=project#sevenbridges.models.project.Project.create)

Reply:
The beautiful thing about this setup (_i.e._ giving the hook both the `client` and `tasks` properties) is that if the `tasks` object doesn't offer a convenience function for a given operation, you can always use the `client` object to make "lower-level" calls. 

For example, `SevenBridgesTasks` doesn't have a convenience method for creating a project, but the `sevenbridges.Api` class does ([docs](https://sevenbridges-python.readthedocs.io/en/latest/quickstart/#project-methods-an-introduction-to-methods-in-the-sevenbridges-python-library)). In an Airflow DAG, it would still be one line of code because the SevenBridges client offers an easy method for creating a project. 

```python
@task
def create_project(task_name, params=None):
    hook = SevenBridgesHook(params["conn_id"])
    project = hook.client.projects.create(name="My new project", billing_group=params["billing_group"])
    return project.id
```

Reply:
Excellent!

Reply:
I posted my comment before seeing yours, @BWMac. Good catch!